test/cpos/test_cpo_connect.cpp

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) Lucian Radu Teodorescu
+ *
+ * Licensed under the Apache License Version 2.0 with LLVM Exceptions
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *   https://llvm.org/LICENSE.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <catch2/catch.hpp>
+#include <execution.hpp>
+#include "test_common/receivers.hpp"
+
+namespace ex = std::execution;
+
+template <typename R>
+struct op_state {
+  int val_;
+  R recv_;
+
+  friend void tag_invoke(ex::start_t, op_state& self) noexcept {
+    ex::set_value((R &&) self.recv_, self.val_);
+  }
+};
+
+struct my_sender {
+  template <template <class...> class Tuple, template <class...> class Variant>
+  using value_types = Variant<Tuple<>>;
+  template <template <class...> class Variant>
+  using error_types = Variant<std::exception_ptr>;
+  static constexpr bool sends_done = false;
+
+  int value_{0};
+
+  template <typename R>
```

Comment:
```suggestion
  template <receiver_of R>
```

Reply:
Created a new sender without `receiver_of` to expose the problem of connecting to non-receivers, and keep the test the same way.